### PR TITLE
Drop use of shapeOfTheEarth for Grib 1

### LINKED ISSUE #241: Fix the assumed spherical earth radius for GRIB1 data.
Possibly, this has always been wrong.
The gribapi has recently changed on that assumption : see [this comment](https://github.com/SciTools/iris-grib/pull/240#issuecomment-762972135).

As it's a breaking change, and come quite late, we included [a fix to preserve the 'old' behaviour](https://github.com/SciTools/iris-grib/pull/240/commits/6474926d86be4191864e21e9aa88ec1d9b2dc396) for release v0.16,.

It might in fact be "more correct" to use the gribapi-provided default 'radiusOfTheEarth' parameter,
so **this needs re-considering.**

### PR DESCRIPTION
**Pull request by lbdreyer**
Fixes #241 

A more detailed summary is on that issue, in [this comment](https://github.com/SciTools/iris-grib/issues/241#issuecomment-1239069695) but to summarise:
In eccodes v2.19.0 the default shapeOfTheEarth was [changed to 0](https://github.com/ecmwf/eccodes/commit/4808994174d735008e70c4b032d447369362ce92), with a radius of 6367470 m. This alligned the behaviour with GRIB2. It was also recommended to *not* use this key and instead use the radius key.

Previously, for backwards compatibility, we [decided to circumvent this change](https://github.com/SciTools/iris-grib/commit/06534bc878c9677fdef7e0a7cd982115004d440c) and hardcode a value of 6 and associate this with the old radius rather than the new radius.

We have now decided to adopt the change in eccodes. This means we should also not use the shapeOfTheEarth key.

We *could* use the radius key instead but this assumes a spherical Earth. I think it would make more sense to use the resolutionAndComponentFlag which distinguishes between a spherical Earth and an oblate spheroidal Earth.

I have added support for a spherical Earth, but not an oblate spheroidal Earth, as this would mean adding new functionality. We could consider adding support for oblate spheroidal later should someone request it.

### CODE REVIEW

File: iris_grib/tests/results/integration/load_convert/sample_file_loads/lambert_grib1.cml [hunk 1]
```diff
@@ -16,16 +16,16 @@
         <auxCoord id="61bde96d" long_name="originating_centre" points="[	US National Weather Service, National Centres for Environmental Prediction]" shape="(1,)" units="Unit('no_unit')" value_type="string"/>
       </coord>
       <coord datadims="[1]">
-        <dimCoord id="62845c7c" points="[-2396487.0124, -2392487.0124, -2388487.0124,
```
Comment (pp-mo) [inline iris_grib/tests/results/integration/load_convert/sample_file_loads/lambert_grib1.cml:19]: I see that the projected transforms (here + polar-stereo) have got changed coordinate values.  I think this is ok/correct -- just a scaling factor due to changed earth radius.

**Review by pp-mo — CHANGES_REQUESTED** [review]
After some background reading on GRIB1, I do agree that this all makes sense.
I think it's right to say that previous behaviour was "wrong" + this is therefore a fix.  
Likewise, the [shift in coordinate values with projected transforms](https://github.com/SciTools/iris-grib/pull/316/files#r1063345620) is I think just a consequence of the adjusted radius -- i.e. effectively a scaling factor, which again is a "bug fix" not "new behaviour".

So, I think all it now just needs is **a whats-new entry** :  highlight the backwards-incompatible changes + explain it is a bug-fix.

File: iris_grib/tests/results/integration/load_convert/sample_file_loads/lambert_grib1.cml (hunk 1 above)
Comment (lbdreyer) [reply iris_grib/tests/results/integration/load_convert/sample_file_loads/lambert_grib1.cml:19]: Yes, I was under a similar impression

**Review by pp-mo — APPROVED** [review]
That sorts it, good to go!

**Pull request merged**